auth-api/src/main/java/io/stargate/auth/api/resources/AuthResource.java

```diff
@@ -232,16 +252,53 @@ public Response createTokenFromUsername(
     } catch (UnauthorizedException e) {
       return Response.status(Response.Status.UNAUTHORIZED)
           .entity(new Error("Failed to create token: " + e.getMessage()))
+          .cacheControl(cacheControlNoStore)
           .build();
     } catch (Exception e) {
       logger.error("Failed to create token", e);
       return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
           .entity(new Error("Failed to create token: " + e.getMessage()))
+          .cacheControl(cacheControlNoStore)
           .build();
     }
 
     return Response.status(Response.Status.CREATED)
         .entity(new AuthTokenResponse().authToken(token))
+        .cacheControl(cacheControlWithMaxAge)
         .build();
   }
+
+  /**
+   * Used to create a Cache-Control header containing the proper values for the token's max-age and
+   * s-maxage as defined by {@code stargate.auth_tokenttl} minus some time to avoid expiration. If
+   * the property is not set then the default value of 1790 will be used.
+   *
+   * @return A {@link CacheControl} object which is converted into a Cache-Control header with
+   *     proper directives.
+   */
+  private CacheControl cacheControlWithAge() {
+    // For a large enough TTL it's safe to knock off 10 seconds but for a smaller number reducing by
+    // that much will negatively impact caching so instead subtract only 1 second.
+    int maxAge = tokenMaxAge <= 100 ? tokenMaxAge - 1 : tokenMaxAge - 10;
```

Comment:
nit: if `tokenMaxAge` is `1` or `0` the result is probably going to be not what we want... WDYT?

Reply:
In either case the maxAge set in the header will be `0` and `-1` respectively which essentially mean the same thing, don't cache. If you have a token TTL of 1 second I think telling it not to cache is valid. As for `0` that's not a valid value and since we're using the same system property on the table based  side you'll have other issues if you set ttl to 0 :)